src/pu-mount.c

```diff
@@ -43,56 +42,42 @@ pu_create_mount_point(const gchar *name,
 static GPtrArray *
 pu_find_mounted(const gchar *device)
 {
-    blkid_probe pr;
-    blkid_partlist ls;
-    gint i;
-    gint num_partitions;
+    struct libmnt_context *ctx;
     struct libmnt_table *tb;
+    struct libmnt_iter *itr;
+    struct libmnt_fs *fs;
+    g_autofree gchar *pattern;
     g_autoptr(GPtrArray) mounted = g_ptr_array_new_with_free_func(g_free);
-    g_autoptr(GError) error = NULL;
 
     g_return_val_if_fail(g_strcmp0(device, "") > 0, NULL);
 
-    pr = blkid_new_probe_from_filename(device);
-    if (!pr)
+    ctx = mnt_new_context();
+    if (!ctx)
         return NULL;
 
-    ls = blkid_probe_get_partitions(pr);
-    if (!ls) {
-        blkid_free_probe(pr);
-        return NULL;
-    }
-    num_partitions = blkid_partlist_numof_partitions(ls);
-    if (num_partitions < 0) {
-        blkid_free_probe(pr);
+    if (mnt_context_get_mtab(ctx, &tb)) {
+        mnt_free_context(ctx);
         return NULL;
     }
-    blkid_free_probe(pr);
-
-    g_debug("Device '%s' has %d partitions", device, num_partitions);
 
-    tb = mnt_new_table();
-    if (!tb)
-        return NULL;
-
-    if (mnt_table_parse_mtab(tb, NULL) < 0) {
-        mnt_unref_table(tb);
+    itr = mnt_new_iter(MNT_ITER_FORWARD);
+    if (!itr) {
+        mnt_free_context(ctx);
         return NULL;
     }
 
-    for (i = 0; i < num_partitions; i++) {
-        g_autofree gchar *partname = pu_device_get_partition_path(device, i + 1, &error);
-        if (!partname) {
-            mnt_unref_table(tb);
-            return NULL;
-        }
-        if (mnt_table_find_source(tb, partname, MNT_ITER_FORWARD)) {
-            g_debug("Partition '%s' is mounted", partname);
-            g_ptr_array_add(mounted, g_steal_pointer(&partname));
+    pattern = g_strdup_printf("^%s[a-z]", device);
```

Comment:
This pattern does not work.

I'd prefer a new utility function being used here, similar to `pu_device_get_partition_path()`:
```suggestion
pattern = pu_device_get_partition_pattern(device, NULL);
```
And implement it like so:
```c
gchar *
pu_device_get_partition_pattern(const gchar *device,
                                GError **error)
{
    g_return_val_if_fail(g_strcmp0(device, "") > 0, NULL);
    g_return_val_if_fail(error == NULL || *error == NULL, NULL);

    if (g_regex_match_simple("(mmcblk|loop)[0-9]+", device, 0, 0)) {
        return g_strdup_printf("%sp[0-9]+", device);
    } else if (g_regex_match_simple("sd[a-z]+", device, 0, 0)) {
        return g_strdup_printf("%s[0-9]+", device);
    } else {
        g_set_error(error, PU_ERROR, PU_ERROR_FAILED,
                    "Invalid device name '%s'", device);
        return NULL;
    }
}
```